### PR TITLE
feat(CX-1892): Introduce new "New Works for You" rail

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

Resolves [CX-1892]

### Description

Implements new "New Works For You" home screen rail behind a feature flag. If the feature flag is turned on, the rail replaces the "New Works By Artists You Follow" rail.

Note: This is the first iteration and probably there will be changes to the copy of the rail.

| | |
| --- | --- |
|simulator_screenshot_23FC934F-3C57-400A-8C75-D56DBB1EED86 | simulator_screenshot_2B0BAB00-80C0-4D51-9B99-D91C3F5D59EA|

[Tech Plan](https://www.notion.so/artsy/App-Homefeed-Make-Implicit-Collector-Interest-Available-44010887c7074d3eac6a6a9bdd9f8c68)

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Add new "New Works For You" rail (behind feature flag) - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[CX-1816]: https://artsyproduct.atlassian.net/browse/CX-1816

[CX-1892]: https://artsyproduct.atlassian.net/browse/CX-1892

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b450d492da569bae3a623e179d70eeadf608effa
</p>

File: src/lib/Scenes/Home/Components/ArtworkRail.tsx [hunk 1]
```diff
@@ -60,7 +70,17 @@ const ArtworkRail: React.FC<{ rail: ArtworkRail_rail } & RailScrollProps> = ({ r
   }
   // This is to satisfy the TypeScript compiler based on Metaphysics types.
   const artworks = compact(rail.results ?? [])
-  const tracking = useTracking()
+
+  const isHiddenFollowedArtistsRail = rail.key === "followed_artists" && showNewNewWorksForYouRail
+  const showRail = artworks.length && !isHiddenFollowedArtistsRail
```
Comment (olerichter00) [inline src/lib/Scenes/Home/Components/ArtworkRail.tsx:75]: Hiding the artwork rail if either it is the old "New Works By Followed Artists" rail (with enabled feature flag) or they're not artworks to show

File: src/lib/Scenes/Home/Components/SaleArtworksHomeRail.tsx [hunk 2]
```diff
@@ -20,6 +20,8 @@ interface Props {
 }
 
 export const SaleArtworksHomeRail: React.FC<Props> = ({ me, relay, onShow, onHide }) => {
+  const [isLoading, setIsLoading] = useState(false)
```
Comment (olerichter00) [inline src/lib/Scenes/Home/Components/SaleArtworksHomeRail.tsx:23]: Moving this up above the conditional return in order to avoid runtime errors.

**Comment by olerichter00** [comment]
#squashongreen

**Pull request merged**